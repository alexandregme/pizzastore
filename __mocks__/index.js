export const TOPPINGS = [{
  name:"topping 1",
  price:1,
  isChecked:true
},{
  name:"topping 2",
  price:2,
  isChecked:false
},{
  name:"topping 3",
  price:3,
  isChecked:false
},{
  name:"topping 4",
  price:4,
  isChecked:false
},{
  name:"topping 5",
  price:5,
  isChecked:false
},{
  name:"topping 6",
  price:6,
  isChecked:false
},{
  name:"topping 7",
  price:7,
  isChecked:false
},{
  name:"topping 8",
  price:8,
  isChecked:false
}];

export const DEFAULT_TOPPING = {
  name:"default topping",
  price:0,
  isChecked:false
};

export const CUSTOM_TOPPING = {
  name:"custom topping",
  price:1,
  isChecked:true
};

export const DEFAULT_PIZZA = {
  "name": "default pizza",
  "maxToppings": 0,
  "basePrice": 0,
  "totalPrice": 0,
  toppings:[]
};

export const CUSTOM_PIZZA = {
  "name": "custom pizza",
  "maxToppings": 1,
  "basePrice": 1,
  "totalPrice": 1,
  toppings:[{
    name:"topping 1",
    price:1,
    isChecked:true
  }]
};

export const PIZZAS = [{
  "name": "small",
  "maxToppings": 3,
  "basePrice": 9.89,
  toppings: TOPPINGS
},{
  "name": "medium",
  "maxToppings": 5,
  "basePrice": 10.89,
  toppings: TOPPINGS
}, {
  "name": "large",
  "maxToppings": null,
  "basePrice": 13.49,
  toppings: TOPPINGS
}];
