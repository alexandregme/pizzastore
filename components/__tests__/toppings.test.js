import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import Toppings from '../toppings';
import Topping from "../topping";
import { TOPPINGS } from "../../__mocks__";

describe("Toppings", ()=>{
  let mountedToppings;
  beforeEach(()=>{
    mountedToppings = shallow(<Toppings />);
  });

  it('renders without crashing', () => {
    shallow(<Toppings />);
    expect(mountedToppings).toMatchSnapshot();
  });

  it('expect to show no pizzas found msg', () => {
    let notFoundMsg = mountedToppings.find('p');
    expect(notFoundMsg).toHaveLength(1);
    expect(notFoundMsg.text()).toEqual('There are no toppings available for this pizza.');
  });
});

describe("Toppings Custom component props", ()=>{
  let mountedToppings;
  let props = {
    toppings: TOPPINGS
  };

  beforeEach(()=>{
    mountedToppings = shallow(<Toppings {...props} />);
  });

  it('expect to render 7 pizza sizes', () => {
    let toppings = mountedToppings.find(Topping);
    expect(toppings.length).toEqual(7);
  });
});
