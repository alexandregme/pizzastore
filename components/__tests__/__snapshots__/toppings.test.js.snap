// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toppings renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Toppings />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        <p>
          There are no toppings available for this pizza.
        </p>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          There are no toppings available for this pizza.
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "There are no toppings available for this pizza.",
        },
        "ref": null,
        "rendered": "There are no toppings available for this pizza.",
        "type": "p",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <p>
            There are no toppings available for this pizza.
          </p>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            There are no toppings available for this pizza.
          </p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "There are no toppings available for this pizza.",
          },
          "ref": null,
          "rendered": "There are no toppings available for this pizza.",
          "type": "p",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
